src/panels/config/integrations/integration-panels/bluetooth/bluetooth-device-page.ts

```diff
@@ -0,0 +1,160 @@
+import type { CSSResultGroup, PropertyValues, TemplateResult } from "lit";
+import { html, LitElement } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import memoizeOne from "memoize-one";
+import type { UnsubscribeFunc } from "home-assistant-js-websocket";
+import type { HASSDomEvent } from "../../../../../common/dom/fire_event";
+import type { LocalizeFunc } from "../../../../../common/translations/localize";
+import type {
+  DataTableColumnContainer,
+  RowClickedEvent,
+} from "../../../../../components/data-table/ha-data-table";
+import "../../../../../components/ha-fab";
+import "../../../../../components/ha-icon-button";
+import "../../../../../layouts/hass-tabs-subpage-data-table";
+import { haStyle } from "../../../../../resources/styles";
+import type { HomeAssistant, Route } from "../../../../../types";
+import type {
+  DeviceRowData,
+  EventDataMessage,
+} from "../../../../../data/bluetooth";
+
+import { subscribeBluetoothAdvertisements } from "../../../../../data/bluetooth";
+import { showBluetoothDeviceInfoDialog } from "./show-dialog-bluetooth-device-info";
+
+@customElement("bluetooth-device-page")
+export class BluetoothDevicePage extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public route!: Route;
+
+  @property({ type: Boolean }) public narrow = false;
+
+  @property({ attribute: "is-wide", type: Boolean }) public isWide = false;
+
+  @state() private _data: { [id: string]: DeviceRowData } = {};
+
+  private _unsub?: Promise<UnsubscribeFunc>;
+
+  private _firstUpdatedCalled = false;
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    if (this.hass && this._firstUpdatedCalled) {
+      this._unsub = subscribeBluetoothAdvertisements(
+        this.hass,
+        this._handleIncomingEventDataMessage.bind(this)
+      );
+    }
+  }
+
+  protected firstUpdated(changedProperties: PropertyValues): void {
+    super.firstUpdated(changedProperties);
+    if (this.hass) {
+      this._unsub = subscribeBluetoothAdvertisements(
+        this.hass,
+        this._handleIncomingEventDataMessage.bind(this)
+      );
+      this._firstUpdatedCalled = true;
+    }
+  }
+
+  private _handleIncomingEventDataMessage(event: EventDataMessage) {
+    if (event.add) {
+      for (const device_data of event.add) {
+        this._data[device_data.address] = device_data;
```

Comment:
You should copy the object before making changes to ensure that Lit sees it as changed data.